deepchem/molnet/load_function/load_dataset_template.py

```diff
@@ -0,0 +1,198 @@
+MYDATASET_CSV_URL = 'http://deepchem.io.s3-website-us-west-1.amazonaws.com/datasets/mydataset.csv'
+
+
+def load_mydataset(featurizer: str = None,
+                   split: str = 'random',
+                   reload: bool = True,
+                   move_mean: bool = True,
+                   data_dir: str = None,
+                   save_dir: str = None,
+                   **kwargs) -> Iterable:
+  """Load mydataset.
+
+  This is a template for adding a function to load a dataset from
+  MoleculeNet. Adjust the global variable URL strings, default parameters,
+  and variable names as needed. The function will need to be modified
+  to handle the allowed featurizers for your dataset. 
+
+  If `reload = True` and `data_dir` (`save_dir`) is specified, the loader
+  will attempt to load the raw dataset (featurized dataset) from disk.
+  Otherwise, the dataset will be downloaded from the DeepChem AWS bucket.
+
+  The dataset will be featurized with `featurizer` and separated into
+  train/val/test sets according to `split`. Additional kwargs may
+  be given for specific featurizers and splitters.
+
+  Please refer to the MoleculeNet documentation for further information
+  https://deepchem.readthedocs.io/en/latest/moleculenet.html.
+  
+  Parameters
+  ----------
+  featurizer: {List of allowed featurizers for this dataset}
+    A featurizer that inherits from deepchem.feat.Featurizer.
+  split: {'random', 'stratified', 'index', 'scaffold'}
+    A splitter that inherits from deepchem.splits.splitters.Splitter.
+  reload: bool (default True)
+    Try to reload dataset from disk if already downloaded. Save to disk
+    after featurizing.
+  move_mean: bool (default True)
+    Center data to have 0 mean after transform.
+  data_dir: str, optional
+    Path to datasets.
+  save_dir: str, optional
+    Path to featurized datasets.
+  **kwargs: optional arguments to featurizers and splitters.
+
+  Returns
+  -------
+  tasks, datasets, transformers : iterable
+    tasks : list
+      Column names corresponding to machine learning target variables.
+    datasets : tuple
+      train, validation, test splits of data as
+      ``deepchem.data.datasets.Dataset`` instances.
+    transformers : list
+      ``deepchem.trans.transformers.Transformer`` instances applied
+      to dataset.
+
+  References
+  ----------
+  MLA style references for this dataset. E.g.
+    Wu, Zhenqin et al. "MoleculeNet: a benchmark for molecular
+      machine learning." Chemical Science, vol. 9, 2018, 
+      pp. 513-530, 10.1039/c7sc02664a.
+
+    Last, First et al. "Article title." Journal name, vol. #,
+      no. #, year, pp. page range, DOI. 
+
+  Examples
+  --------
+  >>> import deepchem as dc
+  >>> tasks, datasets, transformers = dc.molnet.load_mydataset()
+  >>> train_dataset, val_dataset, test_dataset = datasets
+  >>> n_tasks = len(tasks)
+  >>> n_features = train_dataset.get_data_shape()[0]
+  >>> model = dc.models.MultiTaskClassifier(n_tasks, n_features)
+
```

Comment:
Can you add a 

```
Returns
--------
...
```
section that describes the type of the expected return values?

Reply:
The `Returns` section is a bit further up in the docstring, here:
https://github.com/ncfrey/deepchem/blob/e8999c1386970cb1de1835c245d4f3d28b8e8c5d/deepchem/molnet/load_function/load_dataset_template.py#L60-L64

I think the ordering is following numpy conventions, but I'm not sure and it may be confusing to not have `Returns`  as the last section.

Reply:
Oh my bad! Missed this in my review pass.